source/base/parameter_handler.cc

```diff
@@ -1382,10 +1382,9 @@ bool ParameterHandler::read_input (const std::string &filename,
   try
     {
       std::string openname = search.find(filename);
-      std::ifstream input (openname.c_str());
-      AssertThrow(input, ExcIO());
-
-      return read_input (input, filename);
+      std::ifstream file (openname.c_str());
+      AssertThrow(file, ExcIO());
+      return read_input_from_string (openname.c_str());
```

Comment:
I don't think this would work. Did you test this? You pass the *file name*, not the *content of the file* to `read_input_from_string`. My idea was that you simply read the entire file into a string (basically like you did in the first round of review, but without the processing) and then pass that string to `read_input_from_string()`.